### PR TITLE
Remove unused operator methods and impl

### PR DESCRIPTION
**Pull request by dhruvmanila**
## Summary

This PR removes unused operator methods and impl traits. There is already the `is_macro::Is` implementation for all the operators and this seems unnecessary.

### CODE REVIEW

**Review by AlexWaygood — APPROVED** [review]
Makes sense to me (but I'm still relatively new to working on this code)

File: crates/ruff_python_ast/src/nodes.rs [hunk 1]
```diff
@@ -2727,125 +2727,13 @@ pub enum ExprContext {
...
-    #[inline]
-    pub const fn del(&self) -> Option<ExprContextDel> {
-        match self {
-            ExprContext::Del => Some(ExprContextDel),
-            _ => None,
-        }
-    }
-}
-
-pub struct ExprContextLoad;
```
Comment (MichaReiser) [inline crates/ruff_python_ast/src/nodes.rs:2756]: Hum, the `ExprContext*` star types were interesting... Why did they exist when `ExprContext::Load`, `::Store` etc. exits too. Anyway, it makes sense to remove them

**Review by MichaReiser — APPROVED** [review]
It seems we inherited all these `struct` types from `RustPython`. It's unclear to me why they existed.

File: crates/ruff_python_ast/src/nodes.rs (hunk 1 above)
Comment (dhruvmanila) [reply crates/ruff_python_ast/src/nodes.rs:2756]: Yeah, I'm not sure either, maybe to convert to a Python class https://github.com/RustPython/Parser/blob/9ce55aefdeb35e2f706ce0b02d5a2dfe6295fc57/ast-pyo3/src/gen/to_py_ast.rs#L987-L997 ?

**Pull request merged**